### ISSUE TITLE
`|&:|` syntax should be obsolete

### ISSUE THREAD

**Issue by japaric** [rust-lang/rust#22881]
`|:|` and `|&mut:|` are obsolete, but `|&:|` is not.

``` rust
fn main() {
    let a = |:| {};  //~ error: obsolete syntax: `:`, `&mut:`, or `&:`
    let b = |&:| {};  // accepted
    let c = |&mut :| {};  //~ error: obsolete syntax: `:`, `&mut:`, or `&:`
}
```

```
rustc 1.0.0-nightly (b47aebe3f 2015-02-26) (built 2015-02-27)
```

**Comment by japaric** [comment]
I think I spotted the problem, testing...

**Comment by japaric** [comment]
Fix in #22884
